lib/arm/arm_lifter.ml

```diff
@@ -320,22 +320,19 @@ let lift_bits mem ops (insn : bits_insn ) =
   (* reverses *)
   | `REV, [|`Reg dest; src; cond; _|] ->
     let s = exp_of_op src in
-    let i24 = int32 24 in
-    let i8 = int32 8 in
-    let umask = int32 0xff0000 in
-    let lmask = int32 0xff00 in
-    let rev =
-      let open Bil in
-      s              lsl i24 lor
-      s              lsr i24 lor
-      (s land umask) lsr i8  lor
-      (s land lmask) lsl i8
-    in
+    let rev = Bil.(concat
```

Comment:
First of all, the code is not indented correctly. Second, and most importantly, you can use the [(^)] operator to make your code much more readable, e.g.,
```ocaml
    let rev = Bil.(extract 7 0 s ^
                   extract 15 8 s ^
                   extract 23 16 s ^
                   extract 31 24 s) in
```

[1]: https://binaryanalysisplatform.github.io/bap/api/master/bap/Bap/Std/Bil/index.html#val-(^)